sdk/confidentialledger/azure-security-confidentialledger/src/test/java/com/azure/security/confidentialledger/ConfidentialLedgerClientTestBase.java

```diff
@@ -41,89 +40,83 @@ class ConfidentialLedgerClientTestBase extends TestBase {
             }
-            confidentialLedgerCertificateClient = confidentialLedgerCertificateClientBuilder
-                    .buildClient();
-            
-            String ledgerId = Configuration.getGlobalConfiguration().get("LEDGER_NAME", "java-sdk-test");
+
+            confidentialLedgerCertificateClient = confidentialLedgerCertificateClientBuilder.buildClient();
+
+            String ledgerName = Configuration.getGlobalConfiguration().get("LEDGER_NAME", "emily-java-sdk-tests");
+
+            Response<BinaryData> ledgerIdentityWithResponse = confidentialLedgerCertificateClient
+                .getLedgerIdentityWithResponse(ledgerName, null);
+            BinaryData identityResponse = ledgerIdentityWithResponse.getValue();
+            ObjectMapper mapper = new ObjectMapper();
+            JsonNode jsonNode = null;
+            try {
+                jsonNode = mapper.readTree(identityResponse.toBytes());
+            } catch (IOException ex) {
+                System.out.println("Caught exception " + ex);
+                Assertions.assertTrue(false);
+            }
+
+            String ledgerTlsCertificate = jsonNode.get("ledgerTlsCertificate").asText();
+
+            reactor.netty.http.client.HttpClient reactorClient = null;
+
+            try {
+                SslContext sslContext = SslContextBuilder.forClient()
+                    .trustManager(new ByteArrayInputStream(
+                        ledgerTlsCertificate.getBytes(StandardCharsets.UTF_8)))
+                    .build();
+                reactorClient = reactor.netty.http.client.HttpClient.create()
+                    .secure(sslContextSpec -> sslContextSpec.sslContext(sslContext));
+            } catch (SSLException ex) {
+                System.out.println("Caught exception " + ex);
+                Assertions.assertTrue(false);
+            }
+
+            HttpClient httpClient = new NettyAsyncHttpClientBuilder(reactorClient).wiretap(true).build();
 
             confidentialLedgerClientBuilder = new ConfidentialLedgerClientBuilder()
-                    .ledgerEndpoint(Configuration.getGlobalConfiguration().get("LEDGER_URI", "https://emily-java-sdk-tests.confidential-ledger.azure.com"))
-                    .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC));
+                .ledgerEndpoint(Configuration.getGlobalConfiguration().get("LEDGER_URI", "https://emily-java-sdk-tests.confidential-ledger.azure.com"))
+                .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC));
+
             if (getTestMode() == TestMode.PLAYBACK) {
-                confidentialLedgerClientBuilder
-                        .httpClient(interceptorManager.getPlaybackClient())
-                        .credential(request -> Mono.just(new AccessToken("this_is_a_token", OffsetDateTime.MAX)));
+                confidentialLedgerClientBuilder = confidentialLedgerClientBuilder
```

Comment:
You don't need to reassign the builder variable to itself, that's the beauty of builders, you can modify them on the spot :)
```suggestion
                confidentialLedgerClientBuilder
```

Reply:
Done!